packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -91,6 +116,20 @@ function ProviderOptionsScreen({ route, navigation }: Props) {
     })
   }, [])
 
+  const fetchResponse = useAsync(fetchUserIpAddress, [])
+
+  useEffect(() => {
```

Comment:
Instead of using this `useEffect` we could handle all this on the `useAsync` I think? Something like:
```
useAsync(async () => {
  const { result, status } = await fetchUserIpAddress()

 // All the rest
}, [])
```

Reply:
Yup, this is more contained and much cleaner. Thanks for the suggestion. It also makes it clearer what happens if/when there is an API failure.